srcMain/main.cpp

```diff
@@ -228,16 +228,20 @@ void shasta::main::assemble(
 
 
 
-    // Create the run the output directory. If it exists, stop.
-    if(filesystem::exists(assemblerOptions.commandLineOnlyOptions.assemblyDirectory)) {
-        throw runtime_error(
-            "Assembly directory " +
-            assemblerOptions.commandLineOnlyOptions.assemblyDirectory +
-            " already exists.\n"
-            "Remove it or use --assemblyDirectory to specify a different assembly directory.");
+    // Create the run output directory. If it exists and is not empty then stop.
+    bool dirExists = filesystem::isDirectory(assemblerOptions.commandLineOnlyOptions.assemblyDirectory);
```

Comment:
If the specified `assemblyDirectory` exists but is a file rather than a directory, `isDirectory` returns `false` and the code below will attempt to create a directory with that same name, which will fail because of existing file with the same name. Perhaps it is best to check for that explicitly.

Reply:
Cool